aurellion/src/screens/jobs/components/jobMenu.tsx

```diff
@@ -21,24 +25,27 @@ const Menu = () => {
     { label: "Driver", value: "driver", accessibilityLabel: "Driver" },
   ];
 
+  const fetchAndSetJourneys = async () => {
+    let journeys: Journey[] = [];
+    if (userType === "customer") {
+      journeys = await fetchCustomersJobsObj();
+    } else if (userType === "driver") {
+      journeys = await fetchDriverUnassignedJourneys();
+    }
+    setJobsObjs(journeys);
```

Comment:
I dont know how await will work with this piece of code as it might try to complete synchronously before the if as it is in the same scope as it. it may be better to change this to be set directly in the scope of the await so you can guarantee it wont run before them. just an idea:
```suggestion
    if (userType === "customer") {
      journeys = await fetchCustomersJobsObj();
      setJobsObjs(journeys);
    } else if (userType === "driver") {
      journeys = await fetchDriverUnassignedJourneys();
      setJobsObjs(journeys);
    }
```

Reply:
I don't think this is necessary. As we are using await, it will pause execution till the promise is resolved within the if condition, so I don't think there will be an issue

Reply:
yes but the issue was it being ran before the if statement since the code runs asynchronously this could be an issue no?